astropy/coordinates/representation.py

```diff
@@ -958,25 +1220,29 @@ def from_cartesian(cls, cart):
 
     def represent_as(self, other_class):
         # Take a short cut if the other class is a spherical representation
-        if issubclass(other_class, PhysicsSphericalRepresentation):
-            return other_class(phi=self.lon, theta=90 * u.deg - self.lat, r=1.0,
-                               copy=False)
-        elif issubclass(other_class, SphericalRepresentation):
-            return other_class(lon=self.lon, lat=self.lat, distance=1.0,
-                               copy=False)
-        else:
-            return super(UnitSphericalRepresentation,
-                         self).represent_as(other_class)
+        if inspect.isclass(other_class):
+            if issubclass(other_class, PhysicsSphericalRepresentation):
+                return other_class(phi=self.lon, theta=90 * u.deg - self.lat, r=1.0,
```

Comment:
This drops the differential! This may actually not be a bad idea, but it is not what the default implementation does.